src/Catalog/CatalogService.php

```diff
@@ -17,6 +17,62 @@
 
 final class CatalogService extends AbstractService implements CatalogServiceInterface
 {
+    public function getJsonExportProducts(int $page = 1): string
+    {
+        $response = $this->client->get("catalog/export/{$page}");
+
+        if (!isset($response['result'])) {
+            throw new \Exception('An error occured while attempting to retrieve the products.');
+        }
+
+        return $response['result'];
+    }
+
+    /**
+     * @return ProductSummary[]
+     */
+    public function getAllProducts(): array
+    {
+        $products = [];
```

Comment:
plutôt que d'avoir une méthode qui renvoie tous les produits, il serait préférable d'avoir un itérateur sur les produits d'une page, et un itérateur d'itérateur sur les produits de toutes les pages.

Reply:
ok fixed j'ai utilisé un generator plutôt